src/services/tx/txMonitor.ts

```diff
@@ -33,3 +36,52 @@ export const waitForTx = async (provider: JsonRpcProvider, txId: string, txHash:
     })
   }
 }
+
+export const waitForRelayedTx = (taskId: string, txId: string): void => {
+  const gelato = new GelatoRelay()
+
+  const checkTxStatus = async () => {
+    const taskStatus = await gelato.getTaskStatus(taskId)
+
+    switch (taskStatus?.taskState) {
+      case 'CheckPending':
+      case 'ExecPending':
+      case 'WaitingForConfirmation':
+        // still pending we set a timeout to check again
+        setTimeout(checkTxStatus, POLLING_INTERVAL)
+        return
+      case 'ExecSuccess':
+        txDispatch(TxEvent.PROCESSED, {
+          txId,
+        })
+        return
+      case 'ExecReverted':
+        txDispatch(TxEvent.REVERTED, {
+          txId,
+          error: new Error(`Relayed transaction reverted by EVM.`),
+        })
+        return
+      case 'Blacklisted':
+        txDispatch(TxEvent.FAILED, {
+          txId,
+          error: new Error(`Relayed transaction was blacklisted by Relay provider`),
+        })
+        return
+      case 'Cancelled':
+        txDispatch(TxEvent.FAILED, {
+          txId,
+          error: new Error(`Relayed transaction was cancelled by Relay provider`),
+        })
+        return
+      case 'NotFound':
+      default:
+        txDispatch(TxEvent.FAILED, {
+          txId,
+          error: new Error(`Relayed transaction was not found.`),
+        })
+        return
+    }
+  }
+
+  setTimeout(checkTxStatus, 2000)
```

Comment:
```suggestion
  setTimeout(checkTxStatus, POLLING_INTERVAL)
```

Reply:
Here we just need to delay a bit fetching the task status as it is not immediately available if we fetch it right away (though the tx is successfully relayed).
![Screenshot 2023-03-07 at 16 03 44](https://user-images.githubusercontent.com/32431609/223461351-61256cba-47c2-4f77-a015-db0eb37364de.png)

Reply:
Can you move the value to a constant with a comment explaining this?

Reply:
Thanks! I've moved the value to a constant and also added a comment.